src/libs/ascent/runtimes/ascent_main_runtime.cpp

```diff
@@ -616,24 +615,77 @@ AscentRuntime::EnsureDomainIds()
     }
 
     int domain_offset = 0;
+    int max_num_domains = num_domains;
+
 #ifdef ASCENT_MPI_ENABLED
     int *domains_per_rank = new int[comm_size];
     MPI_Allgather(&num_domains, 1, MPI_INT, domains_per_rank, 1, MPI_INT, mpi_comm);
     for(int i = 0; i < m_rank; ++i)
     {
       domain_offset += domains_per_rank[i];
     }
+    for(int i = 0; i < comm_size; i++)
+    {
+      if(domains_per_rank[i] > max_num_domains)
+        max_num_domains = domains_per_rank[i];
+    }
     delete[] domains_per_rank;
 #endif
+
+    std::unordered_set<int> local_unique_ids;
+
+    int *domain_ids = new int[max_num_domains];
```

Comment:
This new may be mis matched with free if an exception is thrown.

We could  use a conduit node to hold these arrays so they are automatically cleaned up for any error cases.

Reply:
I think I ended up rewriting the code enough that I don't hit this case anymore. But for future reference, is this what you meant?

node n;
int *darray = new int[number];
n["array"].set_external(darray);
if(case) ASCENT_ERROR("info"); //hit this, no delete
delete[] darray;

darray will still get cleaned up by conduit. Is my understanding correct?

Reply:
Here is what I would suggest:

```
Node n;
n.set(DataType::c_int(number));
int *dptr = n.value();
// use dptr
// ptr will be cleaned up when n goes out of scope
```